src/test/operators/join_test_runner.cpp

```diff
@@ -184,16 +195,65 @@ class JoinTestRunner : public BaseTestWithParam<JoinTestConfiguration> {
     };
     // clang-format on
 
-    const auto add_configuration_if_supported = [&](const auto& configuration) {
-      // String vs non-String comparisons are not supported in Hyrise and therefore cannot be tested
-      if ((configuration.data_type_left == DataType::String) != (configuration.data_type_right == DataType::String)) {
-        return;
+    /**
+     * Returns a set of adapted configurations if the join type provides further configuration possibilities
+     * which should be tested.
+     * Otherwise, a vector containing only the passed configuration is returned.
+     */
+    const auto build_join_type_specific_variations = [&](const auto& configuration) {
+      if constexpr (std::is_same_v<JoinOperator, JoinIndex>) {
+        // Currently the Index Join cannot deal with configurations where the join column
+        // data types are different, see Issue #2077
+        if (configuration.data_type_left != configuration.data_type_right) {
+          return std::vector<JoinTestConfiguration>{};
+        }
+
+        // For the JoinIndex, two additional parameters influence its execution behavior.
+        // These are the index side and the availability of indexes on that index side.
+        // Based on the input configuration, two configurations are generated for each possible
+        // index side: configurations (1) with and (2) without indexes on the index side.
```

Comment:
Executing JoinIndex without indexes seems pointless. I'd be more interested in a configuration where the majority of the chunks are indexed, but one chunk is not.